packages/poaps/src/PoapsClient.ts

```diff
@@ -90,4 +103,143 @@ export class PoapsClient {
       nextCursor(poaps.length, limit, offset),
     );
   }
+
+  /**
+   * Gets the secret associated with a specific QR hash.
+   * @param {string} qr_hash - The QR hash.
+   * @returns {Promise<string>} The secret.
+   * @throws {CodeAlreadyClaimedError} If the code is already claimed.
+   * @throws {CodeExpiredError} If the code is expired.
+   */
+  async getCodeSecret(qr_hash: string): Promise<string> {
+    const getCodeResponse = await this.tokensApiProvider.getClaimCode(qr_hash);
+
+    if (getCodeResponse.claimed == true) {
+      throw new CodeAlreadyClaimedError(qr_hash);
+    }
+    if (getCodeResponse.is_active == false) {
+      throw new CodeExpiredError(qr_hash);
+    }
+
+    return getCodeResponse.secret;
+  }
+
+  /**
+   * Retrieves the claim code information for a specific QR hash.
+   * @param {string} qr_hash - The QR hash.
+   * @returns {Promise<GetClaimCodeResponse>} The claim code response.
+   */
+  async getClaimCode(qr_hash: string): Promise<GetClaimCodeResponse> {
+    return await this.tokensApiProvider.getClaimCode(qr_hash);
+  }
+
+  /**
+   * Gets the status of a claim using its unique ID.
+   * @param {string} queue_uid - The unique ID of the claim.
+   * @returns {Promise<MintingStatus>} The status of the claim.
+   */
+  async getClaimStatus(queue_uid: string): Promise<MintingStatus> {
+    const claimStatusResponse = await this.tokensApiProvider.claimStatus(
+      queue_uid,
+    );
+    return claimStatusResponse.status;
+  }
+
+  /**
+   * Waits for a claim's status to move out of the 'IN_PROCESS' or 'PENDING' states.
+   * @param {string} queue_uid - The unique ID of the claim.
+   * @returns {Promise<MintingStatus>} The final status of the claim.
+   */
+  async waitClaimStatus(queue_uid: string): Promise<MintingStatus> {
```

Comment:
```suggestion
  async waitClaimStatus(queue_uid: string): Promise<void> {
```

The wait loops until it arrives to the expected status or throws an error with circuit breaker.

It makes no sense to return the status which is expected.